.semgrep_rules/logprep_specific/processor-rules.yml

```
@@ -0,0 +1,39 @@
+rules:
+  - id: processor-implements-apply-rules-method
+    message: Processor must implement apply_rules method
+    severity: INFO
+    languages: [python]
+    patterns:
+      - pattern: |
+          class $X(RuleBasedProcessor):
+            ...
+      - pattern-not: |
+          class $X(RuleBasedProcessor):
+            def _apply_rules(...):
+              ...
+
+   - id: processor-iterates-over-rules-in-apply-rules-method
+     message: Processor should use the process method to iterate over rules
+     severity: INFO
+     languages: [python]
+     patterns:
+       - pattern: |
+           class $X(RuleBasedProcessor):
+             def _apply_rules(...):
+               ...
+               for rule in ...:
+                 ...
+
+   - id: processor-initialized-without-configuration-dict
+     message: Processor should be initialized with name, configuration and logger
+     severity: INFO
+     languages: [python]
+     patterns:
+       - pattern: |
+           class $X(RuleBasedProcessor):
+             def __init__(self,...):
+               ...
+       - pattern-not: |
+           class $X(RuleBasedProcessor):
+             def __init__(self, name, configuration, logger):
+               ...
```

The indentation for the last two rules must be fixed, otherwise it doesn't run.

```suggestion
  - id: processor-iterates-over-rules-in-apply-rules-method
    message: Processor should use the process method to iterate over rules
    severity: INFO
    languages: [python]
    patterns:
      - pattern: |
          class $X(RuleBasedProcessor):
            def _apply_rules(...):
              ...
              for rule in ...:
                ...

  - id: processor-initialized-without-configuration-dict
    message: Processor should be initialized with name, configuration and logger
    severity: INFO
    languages: [python]
    patterns:
      - pattern: |
          class $X(RuleBasedProcessor):
            def __init__(self,...):
              ...
      - pattern-not: |
          class $X(RuleBasedProcessor):
            def __init__(self, name, configuration, logger):
              ...
```